plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/QueryBuilder.php

```diff
@@ -121,6 +123,26 @@ private function get_valid_query_vars() {
 		return $this->valid_query_vars;
 	}
 
+	/**
+	 * Set the query vars that are used by filter blocks.
+	 *
+	 * @param array $public_query_vars Public query vars.
+	 * @return array
+	 */
+	public function set_query_vars( $public_query_vars ) {
+		$query_vars = array(
+			'status_filter_query_args' => array( ProductFilterStatus::FILTER_STATUS_QUERY_VAR ),
+		);
+
+		return array_reduce(
+			array_values( $query_vars ),
+			function ( $acc, $query_vars_filter_block ) {
+				return array_merge( $query_vars_filter_block, $acc );
+			},
+			$public_query_vars
+		);
```

Comment:
If I understand it correctly, this reduce is equivalent to:

```
foreach ($query_vars as $query_vars_filter_block) {
    $public_query_vars = array_merge($query_vars_filter_block, $public_query_vars);
}
return $public_query_vars;
```

It may be just me, but this reduce looks more complex than it should and replacing with `foreach` would make it easier to read 😅